### ISSUE TITLE
How to measure a performance using confusion matrix?

### ISSUE THREAD

**Issue by horyan971211** [dusty-nv/jetson-inference#1532]
Hi all. I currently running train.py which is from jetson inference. I wish to measure the network performance by using a confusion matrix. Since I am new to this terminology I am not sure how I can get the prediction and true dataset from the model.

Can someone guide me through this or there is a better way to analyze the performance of the model?
How am I going to do that? Any recommendations for coding?

Thank you.

**Comment by dusty-nv** [comment]
Hi @horyan971211 it would need to be with a model that you trained in PyTorch/TensorFlow/ect.   jetson-inference runtime library is for inferencing only and doesn't load the groundtruth dataset, so it wouldn't have that data to compute the confusion matrix.

**Comment by dusty-nv** [comment]
Sure, you could add scikit-learn to this PyTorch train.py script for classification models for example:  https://github.com/dusty-nv/pytorch-classification/blob/dd2548357acb46e376543a3475942075f4a5ce88/train.py

**Comment by horyan971211** [comment]
> Sure, you could add scikit-learn to this PyTorch train.py script for classification models for example: https://github.com/dusty-nv/pytorch-classification/blob/dd2548357acb46e376543a3475942075f4a5ce88/train.py

One more question. Did the jetson inference is for deep learning or transfer learning?

**Comment by dusty-nv** [comment]
By default that train.py script will use transfer learning - it starts with models that were pre-trained on the 1000-class ImageNet dataset.  However you can configure it so that it trains from scratch, but it will take longer to train then.

**Comment by horyan971211** [comment]
> By default that train.py script will use transfer learning - it starts with models that were pre-trained on the 1000-class ImageNet dataset. However you can configure it so that it trains from scratch, but it will take longer to train then.



> Sure, you could add scikit-learn to this PyTorch train.py script for classification models for example: https://github.com/dusty-nv/pytorch-classification/blob/dd2548357acb46e376543a3475942075f4a5ce88/train.py

Can you please provide example?

**Comment by dusty-nv** [comment]
Sorry, I haven't done it.   https://forums.developer.nvidia.com/t/how-to-measure-acc-loss-f1-etc-on-jetson-nano/237878/2
